### ISSUE TITLE
Question about loss functions

### ISSUE THREAD

**Issue by moskomule** [xingruiyu/coteaching_plus#1]
Hi, thank you for publishing the codebase. I have two questions if the loss functions below are designed as you intend.

## 1
In https://github.com/xingruiyu/coteaching_plus/blob/master/loss.py#L77, `loss1` is calculated as 

```
loss_1 = torch.sum(update_step*cross_entropy_1)/labels.size()[0]
```

Here, `cross_entropy_1` is a scalar thus `update_step*cross_entropy_1` is equal to `update_step.sum()*cross_entropy_1` so `update_step` does not work as gates.

## 2
In https://github.com/xingruiyu/coteaching_plus/blob/master/loss.py#L35, the losses are calculated as

```
torch.sum(loss_1_update)/num_remember
```

But since the loss `F.cross_entropy(y_1[ind_2_update], t[ind_2_update])` is already batch-wise averaged, if I understand correctly, the loss is twice devided by batch size.

Thank you again,
Best.

**Comment by hongxin001** [comment]
same question~

**Comment by jisoolee0123** [comment]
Dear authors, thanks for sharing the code. 
Relatedly, I also have a question about loss function. 
In the function “loss_coteaching_plus”, softmax values are used as inputs of "loss_coteaching()".
https://github.com/xingruiyu/coteaching_plus/blob/a68ed612885a258d8fbfe3a594b6474b27e3b575/loss.py#L38
https://github.com/xingruiyu/coteaching_plus/blob/a68ed612885a258d8fbfe3a594b6474b27e3b575/loss.py#L68
And these inputs are used to compute cross entropy loss as follows:
https://github.com/xingruiyu/coteaching_plus/blob/a68ed612885a258d8fbfe3a594b6474b27e3b575/loss.py#L11
However, F.cross_entropy takes logits as inputs, not softmax values. 
Thanks!

**Comment by zhangchuanyi96** [comment]
same question~
Waiting for answers

**Comment by xyy19920105** [comment]
i think this code has some low level error, like author feed the softmax result into the cross_entropy function.

**Comment by LLeiSong** [comment]
I wonder the author give up the method and stop maintaining this repo?

**Comment by randydkx** [comment]
Also about the loss function ,some error exists.

**Comment by Ffmydy** [comment]
@moskomule how to alter the loss? Do you reproduce the results of the paper?

**Comment by moskomule** [comment]
@Ffmydy It's a long ago, so I don't remember. Did you reproduce the results?

**Comment by Ffmydy** [comment]
I didn't.Can you tell me the detail about altering the loss?



---- 回复的原邮件 ----
| 发件人 | Ryuichiro ***@***.***> |
| 日期 | 2022年01月20日 22:31 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [xingruiyu/coteaching_plus] Question about loss functions (#1) |

@Ffmydy It's a long ago, so I don't remember. Did you reproduce the results?

**Comment by moskomule** [comment]
@Ffmydy I just asked if the implementation was correct and am waiting for their reply for 3 years.

**Comment by Ffmydy** [comment]
Thanks for your reply，maybe the author did not reproduce the result of paper.



---- 回复的原邮件 ----
| 发件人 | Ryuichiro ***@***.***> |
| 日期 | 2022年01月20日 22:53 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [xingruiyu/coteaching_plus] Question about loss functions (#1) |

@Ffmydy I just asked if the implementation was correct and am waiting for their reply for 3 years.

**Comment by myyy777** [comment]
I this the wrong is the loss function.
_update_step = np.logical_or(logical_disagree_id, step < 5000).astype(np.float32)
why use the step?
step = epoch * i
I think there's something wrong here.
